e2e/record-types/record-types.e2e-spec.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * This file is part of ng2-json-editor.
+ * Copyright (C) 2017 CERN.
+ *
+ * ng2-json-editor is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * ng2-json-editor is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with ng2-json-editor; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+ * In applying this license, CERN does not
+ * waive the privileges and immunities granted to it by virtue of its status
+ * as an Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+import { RecordTypesPage } from './record-types.po';
+
+describe('Record-Types App', function() {
+  let page: RecordTypesPage;
+  let recordFile: Object;
+  let records = [
+    'authors',
+    'conferences',
+    'experiments',
+    'hep',
+    'institutions',
+    'jobs',
+    'journals'
+  ];
+
+  beforeEach(() => {
+    page = new RecordTypesPage();
+  });
+
+  it('should match the value of a random path in each record with the value of the same path in UI.', () => {
+    for (let index = 0; index < records.length; index++) {
+      let recordType = records[index];
+      recordFile = require(page.getPathForRecordType(recordType));
+      page.navigateTo(`/${recordType}`);
+      let idPath = page.getRandomIdPathFromObject(recordFile);
```

Comment:
Since this function returns the id and the path (and not only the id) the name could be better. You can check http://www.2ality.com/2014/06/es6-multiple-return-values.html and this should work

```javascript
let {path, value} = page.getRandomIdPathAndValueFromObject(record);
```